dpctl/tensor/libtensor/include/kernels/elementwise_functions/pow.hpp

```diff
@@ -0,0 +1,269 @@
+//=== POW.hpp -   Binary function POW                    ------  *-C++-*--/===//
+//
+//                      Data Parallel Control (dpctl)
+//
+// Copyright 2020-2023 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//===---------------------------------------------------------------------===//
+///
+/// \file
+/// This file defines kernels for elementwise evaluation of POW(x1, x2)
+/// function.
+//===---------------------------------------------------------------------===//
+
+#pragma once
+#include <CL/sycl.hpp>
+#include <cstddef>
+#include <cstdint>
+#include <type_traits>
+
+#include "utils/offset_utils.hpp"
+#include "utils/type_dispatch.hpp"
+#include "utils/type_utils.hpp"
+
+#include "kernels/elementwise_functions/common.hpp"
+#include <pybind11/pybind11.h>
+
+namespace dpctl
+{
+namespace tensor
+{
+namespace kernels
+{
+namespace pow
+{
+
+namespace py = pybind11;
+namespace td_ns = dpctl::tensor::type_dispatch;
+namespace tu_ns = dpctl::tensor::type_utils;
+
+template <typename argT1, typename argT2, typename resT> struct PowFunctor
+{
+
+    using supports_sg_loadstore = std::negation<
+        std::disjunction<tu_ns::is_complex<argT1>, tu_ns::is_complex<argT2>>>;
+    using supports_vec =
+        std::negation<std::disjunction<tu_ns::is_complex<argT1>,
+                                       tu_ns::is_complex<argT2>,
+                                       std::is_integral<argT1>,
+                                       std::is_integral<argT2>>>;
+
+    resT operator()(const argT1 &in1, const argT2 &in2)
+    {
+        return std::pow(in1, in2);
```

Comment:
`std::pow()` convert integer inputs to double, see [note](https://en.cppreference.com/w/cpp/numeric/math/pow):
```
Otherwise, if num1 and/or num2 has type double or an integer type, then std::pow(num1, num2) has the same effect as std::pow(static_cast<double>(num1), static_cast<double>(num2)).
```
and as result it will cause a crash on Iris Xe, since no fp64 support there.
We faced the same in dpnp and were trying to resolve by PR [#1463](https://github.com/IntelPython/dpnp/pull/1463).

Reply:
Thank you for pointing this out.

I'll add Vlad's ipow implementation before converting this to ready for review.